### PR TITLE
remove optimizer related codes

### PR DESCRIPTION
**Pull request by daquexian**
Remove optimizer-related codes including `onnx.optimizer`, `onnx.utils.polish_model`, their docs and jupyter notebooks

### CODE REVIEW

File: onnx/utils.py [hunk 1]
```diff
@@ -10,24 +10,11 @@
 
 import onnx.checker
 import onnx.helper
-import onnx.optimizer
 import onnx.shape_inference
 
 from onnx import ModelProto, NodeProto, TensorProto, ValueInfoProto
 
 
-def polish_model(model):  # type: (ModelProto) -> ModelProto
```
Comment (askhade) [inline onnx/utils.py:19]: you can keep this method just remove optimize and the checker call after optimize...

**Review by askhade — APPROVED** [review]

**Review by jcwchen — APPROVED** [review]
LGTM. Thank you for the transfer!

File: onnx/utils.py (hunk 1 above)
Comment (daquexian) [reply onnx/utils.py:19]: If so this method will only run strip_doc_string and shape_inference. It makes this method not so useful. What's your opinion? Thanks!

**Review by linkerzhang — APPROVED** [review]
Thank you very much! Jianhao.

@houseroad please double check this change. The optimizer now is moved to a separate repo under onnx org.

**Review by gramalingam — APPROVED** [review]

**Pull request merged**